src/hooks/useRecoveryTxNotification.ts

```diff
@@ -0,0 +1,78 @@
+import { useEffect } from 'react'
+
+import { formatError } from '@/utils/formatters'
+import { selectNotifications, showNotification } from '@/store/notificationsSlice'
+import { useAppDispatch } from '@/store'
+import useSafeAddress from './useSafeAddress'
+import { RecoveryEvent, RecoveryEventType, recoverySubscribe } from '@/services/recovery/recoveryEvents'
+import { getExplorerLink } from '@/utils/gateway'
+import { useCurrentChain } from './useChains'
+import { useSelector } from 'react-redux'
+
+const RecoveryTxNotifications = {
+  [RecoveryEvent.EXECUTING]: 'Confirm the execution in your wallet.',
+  [RecoveryEvent.PROCESSING]: 'Validating...',
+  [RecoveryEvent.PROCESSED]: 'Successfully validated. Loading...',
+  [RecoveryEvent.REVERTED]: 'Reverted. Please check your gas settings.',
+  [RecoveryEvent.FAILED]: 'Failed.',
+  [RecoveryEvent.SUCCESS]: 'Successfully executed.',
+}
+
+const RecoveryTxNotificationTitles = {
+  [RecoveryEventType.PROPOSAL]: 'Account recovery proposal',
+  [RecoveryEventType.EXECUTION]: 'Account recovery',
+  [RecoveryEventType.SKIP_EXPIRED]: 'Account recovery cancellation',
+}
+
+export function useRecoveryTxNotifications(): void {
+  const dispatch = useAppDispatch()
+  const chain = useCurrentChain()
+  const safeAddress = useSafeAddress()
+
+  const notifications = useSelector(selectNotifications)
+
+  /**
+   * Show notifications of a recovery transaction's lifecycle
+   */
+
+  useEffect(() => {
+    if (!chain?.blockExplorerUriTemplate) {
+      return
+    }
+
+    const unsubFns = Object.entries(RecoveryTxNotifications).map(([event, notification]) =>
+      recoverySubscribe(event as RecoveryEvent, async (detail) => {
+        const isSuccess = event === RecoveryEvent.PROCESSED
+        const isError = 'error' in detail
+
+        const txHash = 'txHash' in detail ? detail.txHash : undefined
```

Comment:
Not sure where to put this comment.
But there is a edge case of having more than one expired txs.
Canceling one will actually cancel all of them. 
Not highest prio as it's an edge case but I think currently it would only mark the first expired recovery proposal as "processing" or being cancelled.

Reply:
It's not possible to propose more than one recovery transaction anymore so I would not concern ourselves about this. Users need to cancel any expired recovery attempts before they can make a new one.